pkg/services/auth/jwt.go

```diff
@@ -0,0 +1,87 @@
+package auth
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/pkg/errors"
+
+	"github.com/weaveworks/weave-gitops/pkg/gitproviders"
+
+	"k8s.io/apimachinery/pkg/util/rand"
+
+	"github.com/golang-jwt/jwt/v4"
+)
+
+//go:generate go run github.com/maxbrunsfeld/counterfeiter/v6 -generate
+
+var SecretKey string
+
+const ExpirationTime = time.Minute * 15
+
+var ErrUnauthorizedToken = errors.New("unauthorized token")
+
+// Claims is a custom JWT claims that contains some token information
+type Claims struct {
+	jwt.StandardClaims
+	Provider      gitproviders.GitProviderName `json:"provider"`
+	ProviderToken string                       `json:"provider_token"`
+}
+
+//counterfeiter:generate . JWTClient
+type JWTClient interface {
+	GenerateJWT(secretKey string, expirationTime time.Duration, providerName gitproviders.GitProviderName, providerToken string) (string, error)
+	VerifyJWT(secretKey string, accessToken string) (*Claims, error)
+}
+
+func NewJwtClient() JWTClient {
+	return &internalJwtClient{}
+}
+
+type internalJwtClient struct {
```

Comment:
I think this should be `internalJWTClient` ?

https://github.com/golang/go/wiki/CodeReviewComments#initialisms